src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -568,6 +573,36 @@ public Boolean[] executeSimple(Object[][] params) {
 		}).description("Returns true if the input is NaN (not a number).")
 			.examples("isNaN(0) # false", "isNaN(0/0) # true", "isNaN(sqrt(-1)) # true")
 			.since("2.8.0");
+
+		Functions.registerFunction(new SimpleJavaFunction<String>("formatNumber", new Parameter[] {
+			new Parameter<>("number", DefaultClasses.NUMBER, true, null),
+			new Parameter<>("format", DefaultClasses.STRING, true, new SimpleLiteral<String>(DEFAULT_NUMBER_FORMAT_STRING, true))
+		}, DefaultClasses.STRING, true) {
+			@Override
+			public String[] executeSimple(Object[][] params) {
+				Number number = (Number) params[0][0];
+				String format = (String) params[1][0];
+
+				if (DEFAULT_NUMBER_FORMAT_STRING.equals(format)) // shortcut
+					return CollectionUtils.array(DEFAULT_NUMBER_FORMAT.format(number));
+
+				try {
+					DecimalFormat numberFormat = NUMBER_FORMAT.get();
+					numberFormat.applyPattern(format);
+					return CollectionUtils.array(numberFormat.format(number));
+				} catch (IllegalArgumentException e) {
+//					Skript.warning("Invalid number format: " + format); // TODO find a better solution for such warnings/errors that doesn't spam the console
+					return null;
+				}
+			}
+		}).description("Converts numbers to human-readable format. By default, '###,###' (e.g. '123,456,789') will be used. For reference, see this "
+		+ "<a href=\"https://docs.oracle.com/javase/7/docs/api/java/text/DecimalFormat.html\" target=\"_blank\">article</a>.")
+			.examples(
+					"command /formatnumber <number>:",
+						"\taliases: fn",
+						"\ttrigger:",
+							"\t\tsend \"Formatted: %formatNumber(arg-1)%\" to sender"
```

Comment:
Some more examples of how to use DecimalFormat would be nice. The table on the javadocs is good, but pretty dense for new users.